### ISSUE TITLE
FSL bet on 7T data

### ISSUE THREAD

**Issue by cementwise** [MINDLAB1/SACE#1]
我在使用FSL bet去颅骨时，在3T上表现尚好，但是在7T数据上经常出现处理后图片破损（7T脑组织部分随颅骨一起被去除）的现象，调整参数并没有太大帮助，请问这个问题有什么好方法吗？

**Comment by MINDLAB1** [comment]
Hi, this is an interesting question. It may be because the resolution for 7T is too high for FSL, or 7T cannot present some brain substructures. 1. If you have paired 3T-7T, do skull-stripping on 3T and then warp the mask to 7T. 2. Down-sampling the 7T and up-sampling the mask.  3. Test other tools, such as SynthiStrip.

**Comment by cementwise** [comment]
还有一个问题，文章图中MAI使用的是IDCT之后的频域图进行卷积，但是代码中逆DCT之后的代码是被注释掉的，训练是应该按照哪种？

**Comment by cementwise** [comment]
还有一个问题，代码中的 self.inv_freq_conv_3x3 = nn.Conv2d(9, 64, 3, 1, 1)被注释掉了，文中得到的形态学特征图是本来在这一步得到的，还是在IDCT之后就已经得到的？

**Comment by MINDLAB1** [comment]
> 还有一个问题，代码中的 self.inv_freq_conv_3x3 = nn.Conv2d(9, 64, 3, 1, 1)被注释掉了，文中得到的形态学特征图是本来在这一步得到的，还是在IDCT之后就已经得到的？

IDCT is implemented by "F.conv2d(input=h_3, weight=self.out_kernel_3x3, padding=1)", where we transform the features from frequency field to spatial field to get morphology features. "self.inv_freq_conv_3x3 = nn.Conv2d(9, 64, 3, 1, 1)" is a "refinement" after IDCT. We did not use it because it is unhelpful for the ultimate performance if I remember correctly.

**Comment by cementwise** [comment]
此外，在CCE中有这样一段：mask = mask / 52 - 1，请问mask/52与-1的目的是什么，从（0，52）映射到（-1，0）吗

**Comment by MINDLAB1** [comment]
> 此外，在CCE中有这样一段：mask = mask / 52 - 1，请问mask/52与-1的目的是什么，从（0，52）映射到（-1，0）吗

Because the mask for brain structures by Freesurfer is [0, 104], mask = mask / 52 - 1 is to map the mask to [-1, 1].

**Comment by cementwise** [comment]
Also, are you using grayscale images? When constructing cosine similarity in SCA module, the cosine similarity of a single channel image is calculated pixel by pixel. Should all results be 1?

**Comment by cementwise** [comment]
此外，您用的是灰度图像吗？在sca 模块中构建余弦相似度时，单通道图像逐像素计算出余弦相似度的话，应该所有相似度都是1，这个您是怎么处理的呢？



发自我的iPhone


------------------ Original ------------------
From: MINDLAB1 ***@***.***&gt;
Date: Fri,Oct 25,2024 10:42 AM
To: MINDLAB1/SACE ***@***.***&gt;
Cc: cementwise ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [MINDLAB1/SACE] FSL bet on 7T data (Issue #1)





  
此外，在CCE中有这样一段：mask = mask / 52 - 1，请问mask/52与-1的目的是什么，从（0，52）映射到（-1，0）吗
  
Because the mask for brain structures by Freesurfer is [0, 104], mask = mask / 52 - 1 is to map the mask to [-1, 1].
